src/modules/Elsa.Http/Middleware/HttpWorkflowsMiddleware.cs

```diff
@@ -0,0 +1,368 @@
+        var workflow = await workflowDefinitionService.FindWorkflowAsync(workflowInstance.DefinitionVersionId, cancellationToken);
+
+        if (workflow == null)
+        {
+            await httpContext.Response.SendNotFoundAsync(cancellation: cancellationToken);
+            return;
+        }
+
+        var workflowState = workflowInstance.WorkflowState;
+        var workflowHostFactory = serviceProvider.GetRequiredService<IWorkflowHostFactory>();
+        var workflowHost = await workflowHostFactory.CreateAsync(workflow, workflowState, cancellationToken);
+        if (await AuthorizeAsync(serviceProvider, httpContext, workflowHost.Workflow, bookmarkPayload, cancellationToken))
+            return;
+
+        var correlationId = await GetCorrelationIdAsync(serviceProvider, httpContext, httpContext.RequestAborted);
+        var cancellationTokens = new CancellationTokens(cancellationToken, cancellationToken);
+        var resumeParams = new ResumeWorkflowHostParams
+        {
+            Input = input,
+            CorrelationId = correlationId,
+            ActivityInstanceId = bookmark.ActivityInstanceId,
+            BookmarkId = bookmark.BookmarkId,
+            CancellationTokens = cancellationTokens
+        };
+
+        var workflowInstanceManager = serviceProvider.GetRequiredService<IWorkflowInstanceManager>();
+        await ExecuteWithinTimeoutAsync(async () =>
+        {
+            await workflowHost.ResumeWorkflowAsync(resumeParams, cancellationToken);
+            await workflowInstanceManager.SaveAsync(workflowHost.WorkflowState, cancellationToken);
+        }, bookmarkPayload.RequestTimeout, httpContext);
+        await HandleWorkflowFaultAsync(serviceProvider, httpContext, workflowHost.WorkflowState, cancellationToken);
+    }
+
+    private async Task<IEnumerable<StoredTrigger>> FindTriggersAsync(IServiceProvider serviceProvider, string bookmarkHash, CancellationToken cancellationToken)
+    {
+        var triggerStore = serviceProvider.GetRequiredService<ITriggerStore>();
+        var triggerFilter = new TriggerFilter
+        {
+            Hash = bookmarkHash
+        };
+        return await triggerStore.FindManyAsync(triggerFilter, cancellationToken);
+    }
+
+    private async Task<IEnumerable<StoredBookmark>> FindBookmarksAsync(IServiceProvider serviceProvider, string bookmarkHash, CancellationToken cancellationToken)
+    {
+        var bookmarkStore = serviceProvider.GetRequiredService<IBookmarkStore>();
+        var bookmarkFilter = new BookmarkFilter
+        {
+            Hash = bookmarkHash
+        };
+        return await bookmarkStore.FindManyAsync(bookmarkFilter, cancellationToken);
+    }
+
+    private async Task ExecuteWithinTimeoutAsync(Func<Task> action, TimeSpan? requestTimeout, HttpContext httpContext)
+    {
+        if (requestTimeout == null)
+        {
+            await action();
+            return;
+        }
+
+        using var cts = new CancellationTokenSource(requestTimeout.Value);
```

Comment:
1. The point of the CTS is to link the existing RequestAborted token with the timeout cancellation token into a new, combine cancellation token that gets triggered when either the RequestAborted or the timeout token is triggered.
2. When the original cancellation token (the original RequestAborted token) is triggered, it will cause the combined cancellation token to be triggered.
3. That's not possible, given that from this point on, no subsequent code will have access to the original cancellation token, which means no code can add more callbacks to it. Only when we install it back into the RequestAborted property of the HTTP context will subsequent code be able to add callbacks to it.